x-pack/plugin/security/cli/src/main/java/org/elasticsearch/xpack/security/cli/AutoConfigInitialNode.java

```diff
@@ -0,0 +1,352 @@
+                // no danger of overwrite because Security was not initially enabled, and this secure setting can only exist in that case
+                nodeKeystore.setString("xpack.security.http.ssl.keystore.secure_password", httpKeystorePassword.getChars());
+            } finally {
+                nodeKeystore.save(env.configFile(), nodeKeystorePassword.getChars());
+            }
+        }
+
+        {
+            Path transportTruststoreOutput = instantAutoConfigDir.resolve(TRANSPORT_AUTOGENERATED_TRUSTSTORE_NAME + ".p12");
+            final KeyStore transportTruststore = KeyStore.getInstance("PKCS12");
+            transportTruststore.load(null);
+            transportTruststore.setCertificateEntry(TRANSPORT_AUTOGENERATED_TRUSTSTORE_NAME, transportCert);
+            fullyWriteFile(transportTruststoreOutput, stream -> transportTruststore.store(stream, new char[0]));
+        }
+
+        {
+            // the truststore contains only the CA certificate
+            // the ES node doesn't strictly require it, but if someone else does, it's good to have it handy
+            // so we don't have to share a keystore with a CA key inside it
+            Path httpTruststoreOutput = instantAutoConfigDir.resolve(HTTP_AUTOGENERATED_TRUSTSTORE_NAME + ".p12");
+            final KeyStore httpTruststore = KeyStore.getInstance("PKCS12");
+            httpTruststore.load(null);
+            httpTruststore.setCertificateEntry(HTTP_AUTOGENERATED_TRUSTSTORE_NAME + "_ca", httpCACert);
+            fullyWriteFile(httpTruststoreOutput, stream -> httpTruststore.store(stream, new char[0]));
+        }
+
+        try (BufferedWriter bw = Files.newBufferedWriter(ymlPath, StandardCharsets.UTF_8, StandardOpenOption.APPEND)) {
+            bw.newLine();
+            bw.newLine();
+            bw.write("###################################################################################");
+            bw.newLine();
+            bw.write("# The following settings, and associated TLS certificates and keys configuration, #");
+            bw.newLine();
+            bw.write("# have been automatically generated in order to configure Security.               #");
+            bw.newLine();
+            bw.write("# These have been generated the first time that the new node was started, on the  #");
+            bw.newLine();
+            bw.write("# terminal, without joining or enrolling to an existing cluster and only if       #");
+            bw.newLine();
+            bw.write("# Security had not been explicitly configured beforehand.                         #");
+            bw.newLine();
+            bw.write(String.format(Locale.ROOT, "# %-79s #", ""));
+            bw.newLine();
+            bw.write(String.format(Locale.ROOT, "# %-79s #", autoConfigDate));
+            bw.newLine();
+            bw.write("###################################################################################");
```

Comment:
mental note to add a link to our docs when we have them. Worth having a TODO here to remind us ?

Reply:
A docs link is a good idea! I'll add a TODO.
Do you know if we have a guidance about doc links in ES output? I mean, is there some sort of special links, that we know they aren't going to move, and maybe be short? (I remember this came up before).